src/lib/GenericPredictor.cc

```diff
@@ -23,25 +23,25 @@ GenericPredictor::GenericPredictor(ryml::ConstNodeRef config)
   btb_ =
       std::vector<std::pair<uint8_t, uint64_t>>(1 << btbBits_, {satCntVal, 0});
```

Comment:
An aside, should we be limiting btbBits_ to max 64 bits?

Reply:
Yes, I think 64 bits makes sense as an upper limit.  Currently the datatype we are using for indexing the btb is uint64_t, so that would have to change (what to?) if we went higher.  Also, a 64bit indexed vector is incredibly large for a btb (certainly larger than is achievable in the real-world)

Reply:
Recall, `btbBits_` purely determines the size of the btb through the calculation 2^btbBits_. A value of 64 would give a btb with 18446744073709552000 entries, which I should hope would be enough.

Potentially you are asking about the number of bits used to store the global history? This made me think what happens if the number of in-flight branches goes above the global history length e.g. this is only set to 19 for the TX2 model which doesn't seem very high considering a ROB of 180 (potentially 36 in-flight branches if they occur 1 in every 5 instructions)

Reply:
This paper https://dl.acm.org/doi/pdf/10.1145/192724.192756 concludes that the accuracy of a speculatively updated global history is unaffected by the number of in-flight instructions.

Reply:
Yes, this idea is good but defer for a later PR

Reply:
Leaving for Jack to mark as resolved